### DISCUSSION TITLE
STL file is working on an AMReX code, but its always running with a refinement of level 2, even if the refinement is set higher

### DISCUSSION THREAD

**Discussion by milin333martin** [AMReX-Codes/amrex#4201] [Q&A]
Hello,

I am trying to use an STL file in an AMReX based code. I am able to initialize the base grid, read the STL file and its parameters, but the refinement criteria for the main grid is automatically brought down to level 2 even if the refinement parameters are set higher in input files are specified.

Here is some of the settings I give in the input files:

### REFINEMENT / REGRIDDING
cns.tag_probspecific  = 1 
amr.max_level         = 4     # maximum level number allowed
cns.refine_max_dengrad_lev = 4
amr.ref_ratio         = 2 2 2 2 # refinement ratio
amr.regrid_int        = 2 # how often to regrid
amr.blocking_factor_x = 16
amr.blocking_factor_y = 8
amr.blocking_factor_z = 8
amr.max_grid_size_x   = 1024
amr.max_grid_size_y	  = 128
amr.max_grid_size_x	  = 128
amr.n_error_buf_x     = 0 # number of buffer cells in error est
amr.n_error_buf_y     = 0
amr.n_error_buf_z     = 0
amr.grid_eff          = 0.98     # what constitutes an efficient grid
amr.refine_grid_layout = true
amr.regrid_on_restart  = 1
amr.compute_new_dt_on_regrid = 1

### EB using STL
eb2.geom_type = stl
eb2.stl_file = ArrestorSTLBigHole.stl
eb2.stl_scale = 0.001  # Adjust as needed
eb2.stl_center = 0.273 0.002 0.001  # Adjust to your desired center
eb2.stl_reverse_normal = 0  # Set to 1 if you need to reverse the normals
eb2.cover_multiple_cuts = 1

The STL is build using the code corresponding to:
```
EB2::Build (const Geometry& geom,
            int required_coarsening_level,
            int max_coarsening_level,
            int ngrow = 4,
            bool build_coarse_level_by_coarsening = true);
```

I dont run into any error or any issues with the initialization of the STL into the domain, but its just the issue with the refinement.

Some guidance to rectify this issue will be greatly appreciated. Thank you in advance.

**Comment by WeiqunZhang** [comment]
Setting `amr.max_level = 4` does not mean there are always 4 refined levels. It depends on whether a cell is tagged for refinement. It seems that the code is based on EB_CNS. If that's the case you should take a look at the `CNS::errorEst` function to see how tagging is done there. Maybe you need to adjust dengrad_threshold. Also, it's completely up to you. You can modify to function to do what you want.
